### ISSUE TITLE
Support custom domain mapping for existing rich integration providers

### ISSUE THREAD

**Issue by oliverdunk** [gitkraken/vscode-gitlens#1943]
I work on a project which has a private Git instance, with the web UI accessible at a different URL to the SSH server. Since GitLens uses the URL configured for the Git remote, it tries to construct URLs by using this as the domain and fails. I'm aware of two ways of working around that:

- Using the `gitlens.remotes` configuration to specify a Custom remote and provide my own URLs. This is the most flexible, but means losing all of the built in config for a provider and having to re-specify every URL.
- Adding a new Git remote with the correct URL (works better, but everyone on the team has to fiddle around with this since it can't be committed as repository configuration).

Is there a better way of doing this? If not, I propose adding an optional config key that allows me to force a specific domain to be passed here, regardless of the matched URL: https://github.com/gitkraken/vscode-gitlens/blob/90bb6ab3c00315eb80acc64aaaf2b6b0d8e80387/src/git/remotes/factory.ts#L90

If this is work you'd be willing to accept, I'd be happy to open a PR.

**Comment by oliverdunk** [comment]
@eamodio Would you be able to take a look at this one and let me know if you'd be open to a PR? :)

**Comment by georgefst** [comment]
Yeah, this would be very nice to have. I've just had to painstakingly recreate the GitHub integration as a custom config. Which also took a while since the [documentation](https://help.gitkraken.com/gitlens/gitlens-settings/#remote-provider-integration-settings) is pretty poor.
